File: 2018/clojure-of-code/test/clojure_of_code/day_01_test.clj
(ns clojure-of-code.day-01-test
  (:require [clojure.test :refer :all]
            [clojure-of-code.day-01 :refer :all]))

(clojure.test/deftest part1
  (testing "part 1"
    (is (= 516 (clojure-of-code.day-01/part-1 (slurp "../../2018/priv/day01/input.txt"))))))

(clojure.test/deftest part2
  (testing "part 2"
    (is (= 71892 (clojure-of-code.day-01/part-2 (slurp "../../2018/priv/day01/input.txt"))))))

File: 2019/elixir/test/advent_of_code_2018/day_01_test.exs
defmodule AdventOfCode2019.Day01Test do
  use ExUnit.Case

  import AdventOfCode2019.Day01

  test "part1" do
    input = [12, 14, 1969, 100_756]
    result = part1(input)

    assert result == 2 + 2 + 654 + 33583
  end

  test "part2" do
    input = [14, 1969, 100_756]
    result = part2(input)

    assert result == 2 + 966 + 50346
  end
end
